src/test/java/com/ms/zookeeper/TestZkUserAuthorizer.java

```diff
@@ -0,0 +1,47 @@
+package com.ms.zookeeper;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+import com.ms.zookeeper.authz.ZkUserAuthorizer;
+
+public class TestZkUserAuthorizer {
+    @Test
+    public void testUserAuth() {
+        ZkUserAuthorizer auth = new ZkUserAuthorizer();
+        assertTrue(auth.authorize("username", "user://username"));
+        assertFalse(auth.authorize("username", "user://otherusername"));
+    }
+
+    @Test
+    public void testUserAuthWithRealms() {
+        System.setProperty("zookeeper.validKerberosRealms", "realm,realm2");
+        ZkUserAuthorizer auth = new ZkUserAuthorizer();
+        assertTrue(auth.authorize("username@REALM", "user://username"));
+        assertTrue(auth.authorize("username@realm", "user://username"));
+        assertFalse(auth.authorize("username@REALM", "user://otherusername"));
+
+        assertTrue(auth.authorize("username@realm2", "user://username"));
+        assertTrue(auth.authorize("username@REALM2", "user://username"));
+        assertFalse(auth.authorize("username@REALM2", "user://otherusername"));
+
+        assertFalse(auth.authorize("username@foo.bar", "user://username"));
+    }
+
+    @Test
+    public void testUserAuthWithHosts() {
+        System.setProperty("zookeeper.validKerberosRealms", "realm,realm2");
+        ZkUserAuthorizer auth = new ZkUserAuthorizer();
+        assertTrue(auth.authorize("host/hostname", "user://host/*"));
```

Comment:
same here - can users and host and everything else be treated same way?

Reply:
the extra function here is the wildcard (*) which we don't have for user. Again an internal feature. Treadmill was using it. Maybe already abandoned and we use only the file authorizer.